ddtrace/mocktracer/mocktracer.go

```diff
@@ -50,38 +49,43 @@ type Tracer interface {
 // interface to query the tracer's state.
 func Start() Tracer {
 	t := newMockTracer()
-	internal.SetGlobalTracer(t)
-	internal.Testing = true
+	tracer.SetGlobalTracer(t)
+	tracer.Testing = true
 	return t
 }
 
 type mocktracer struct {
 	sync.RWMutex  // guards below spans
-	finishedSpans []Span
-	openSpans     map[uint64]Span
+	finishedSpans []*Span
+	openSpans     map[uint64]*Span
 }
 
 func newMockTracer() *mocktracer {
 	var t mocktracer
-	t.openSpans = make(map[uint64]Span)
+	t.openSpans = make(map[uint64]*Span)
 	return &t
 }
 
+// This is called by the spans when they finish
+func (t *mocktracer) FinishSpan(s *tracer.Span) {
+	t.addFinishedSpan(s)
+}
+
 // Stop deactivates the mock tracer and sets the active tracer to a no-op.
 func (*mocktracer) Stop() {
-	internal.SetGlobalTracer(&internal.NoopTracer{})
-	internal.Testing = false
+	tracer.SetGlobalTracer(&tracer.NoopTracer{})
+	tracer.Testing = false
```

Comment:
I find the semantics here a little uncanny... Why can we directly assign to `tracer.Testing` but we have to call a method to set (*waves hands*) `tracer.GlobalTracer`?

It also feels like `tracer.Testing` is only set after `tracer.SetGlobalTracer` is called, so why not have something like:
```suggestion
	tracer.SetGlobalTracer(&tracer.NoopTracer{}, false)
```

Reply:
You're very right, this is gross.

I think this needs to be refactored. Before this was hidden behind an internal package (and was still gross, but at least internally gross). I'd like to put it back behind an internal package or unexport it or something, but at very least I'll change the signature like you suggest.